File: gradle/subprojects/translator/src/main/kotlin/com/app/translator/Translator.kt
package com.app.translator

import kotlin.collections.hashMapOf

val helloTranslations =
    hashMapOf(
        "French" to "Bonjour",
        "Spanish" to "Hola",
        "German" to "Hallo",
        "Italian" to "Ciao",
        "Dutch" to "Hallo",
    )

val greetingsFromTranslations =
    hashMapOf(
        "French" to "salutations de",
        "Spanish" to "saludos desde",
        "German" to "grüße aus",
        "Italian" to "saluti da",
        "Dutch" to "groeten uit",
    )

fun translate(
    language: String,
    name: String,
): String {
    val greeting = helloTranslations[language]
    val greetingsFrom = greetingsFromTranslations[language]

    return if (greeting != null && greetingsFrom != null) {
        "$greeting $name, $greetingsFrom Gradle !"
    } else {
        "Hi $name, greetings from Gradle !"
    }
}

File: dotnet/modules/Translator/Translator.Lib/Translator.cs
namespace Prog.Translation;

public class Translator
{
    private static Dictionary<string, string> HelloTranslations()
    {
        var map = new Dictionary<string, string>
        {
            { "French", "Bonjour" },
            { "Spanish", "Hola" },
            { "German", "Hallo" },
            { "Italian", "Ciao" },
            { "Dutch", "Hallo" }
        };
        return map;
    }

    private static Dictionary<string, string> GreetingsFromTranslations()
    {
        var map = new Dictionary<string, string>
        {
            { "French", "salutations de" },
            { "Spanish", "saludos desde" },
            { "German", "grüße aus" },
            { "Italian", "saluti da" },
            { "Dutch", "groeten uit" }
        };
        return map;
    }

    public static string Translate(string language, string name)
    {
        var helloBinding = HelloTranslations();
        var greetingsFromBinding = GreetingsFromTranslations();

        helloBinding.TryGetValue(language, out string? greeting);
        greetingsFromBinding.TryGetValue(language, out string? greetingsFrom);

        if (greeting != null && greetingsFrom != null)
        {
            return $"{greeting} {name}, {greetingsFrom} .NET !";
        }
        else
        {
            return $"Hi {name}, greetings from .NET !";
        }
    }
}
